lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -124,11 +112,13 @@ public function getExtension()
      */
     protected function getFilename($id)
     {
+        $hash = hash('sha256', $id);
```

Comment:
This has been discussed previously, and it is not really viable due to possible collisions anyway.

This approach can only work if the subclasses store the cache ID inside the stored file, and then check against it on retrieval. Probably the same is to be done also for deletions.